### DISCUSSION TITLE
How to add expand and collapse animation to `admonition`

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#7247] [Q&A] [answered]
How to add expand and collapse animation to `admonition`
I tried to learn CSS, but I couldn't do it well
I tried to use width and other settings, but the effect was not good

**Comment by copdips** [comment]
is this what you want ?
https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=%3F%3F%3F#collapsible-blocks

**Reply by w1ndys** [reply]
Yes, but I want to add a transition animation to the box that expands and collapses when clicked

**Comment by kamilkrzyskow** [comment, marked as answer]
Here is a general solution with pure CSS:
https://dev.to/jgustavoas/solved-how-to-fully-animate-the-details-html-element-with-only-css-and-no-javascript-55ek#result

You just have to adapt it to mkdocs-material

**Reply by w1ndys** [reply]
Very good, very good, this is exactly the effect I wanted.
